tests/drivers/watchdog/wdt_basic_api/src/test_wdt.c

```diff
@@ -300,7 +300,7 @@ static int test_wdt_bad_window_max(void)
 
 void test_wdt(void)
 {
-	if (m_testcase_index != 1) {
+	if (m_testcase_index == 0) {
```

Comment:
This change is not required . As values defined in noinit section need not have default value as zero. On esp32 platform it is observed that  m_testcase_index doesn't default to zero.

Reply:
Is this change is necessary to fix #13468

Reply:
@SavinayDharmappa I have updated the patch, please help me to review it. Thanks!